internals/auth/signature.go

```diff
@@ -62,19 +62,31 @@ type Credential interface {
 	AddAuthentication(r *http.Request) error
 }
 
-// signer contains all necessary credentials to sign a request.
-type signer struct {
-	key crypto.RSAPrivateKey
+// Signer provides proof that given bytes are processed by the owner of the signer.
+type Signer interface {
+	ID() (string, error)
+	Sign([]byte) ([]byte, error)
+	SignMethod() string
 }
 
-// NewRSACredential initializes a new signing credentials struct.
-func NewRSACredential(key crypto.RSAPrivateKey) Credential {
-	return signer{
-		key: key,
+// HTTPSigner proofs that an HTTP request is made by the owner of the signer.
+type HTTPSigner interface {
+	Sign(r *http.Request) error
```

Comment:
What makes you require to call this a `Signer` with a `Sign` function? We used to have something in the `auth.Credential` interface called `AddAuthentication`. Then a token-based credential can also add authentication headers through the same interface. 

What's your view on this?

Reply:
Yes, completely agree. :+1:

Also discussed this with Joris this morning, we want the interface to have generic names (authenticater/authenticate) and the constructor more specific (can stay `NewHTTPSigner`)

Reply:
I've renamed the `HTTPSigner` interface to `Authenticator` with an `Authenticate` function.